keyhandler/keyhandler.c

```diff
@@ -0,0 +1,207 @@
+        printf("%s\n", kvpData);
+
+        // Free the emitted KVP string
+        free(kvpData);
+    }
+}
+
+int main(int argc, char *argv[])
+{
+    int opt;
+    char *profile_file = NULL;
+    char *type = NULL;
+    char *code = NULL;
+    int value;
+    ut_kvp_status_t status;
+    ut_kvp_instance_t *pInstance;
+    struct uinput_setup usetup;
+
+    // Parse command-line arguments
+    /*
+    •	-p <profile_file>
+        Path to the profile YAML file defining event types and key mappings. Required.
+    •	-t <event_type>
+        Type of the event (e.g., EV_KEY). Required.
+    •	-c <event_code>
+        Event code to emit (e.g., KEY_A). Required.
+    •	-v <value>
+        Event value to emit. Required.
+        •	1, 0 etc
+    */
+    while ((opt = getopt(argc, argv, "p:t:c:v:")) != -1)
+    {
+        switch (opt)
+        {
+        case 'p':
+            profile_file = optarg;
+            status = ut_kvp_profile_open(optarg);
+            if (status != UT_KVP_STATUS_SUCCESS)
+            {
+                UT_LOG_ERROR("Failed to Load profile_file [%s]", optarg);
+            }
+            break;
+        case 't':
+            type = optarg;
+            break;
+        case 'c':
+            code = optarg;
+            break;
+        case 'v':
+            value = atoi(optarg);
+            break;
+        default:
+            print_usage(argv[0]);
+            return 1;
+        }
+    }
+
+    // Validate required arguments
+    if (profile_file == NULL)
+    {
+        UT_LOG_ERROR("Error: Missing required arguments.");
+        print_usage(argv[0]);
+        return 1;
+    }
+
+    pInstance = ut_kvp_profile_getInstance();
+    if(pInstance == NULL)
+    {
+        UT_LOG_ERROR("perhaps platform profile was not passed using -p switch");
+        return -1;
+    }
+
+    //print_yaml(pInstance);
+
+    int fd = open("/dev/uinput", O_WRONLY | O_NONBLOCK);
+
+
+    /*
+     * The ioctls below will enable the device that is about to be
+     * created, to pass key events, in this case the space key.
+     */
+    //ioctl(fd, UI_SET_EVBIT, EV_KEY);
+    //ioctl(fd, UI_SET_KEYBIT, result_kvp);
+
+    memset(&usetup, 0, sizeof(usetup));
+    usetup.id.bustype = BUS_USB;
+    usetup.id.vendor = 0x1234;  /* sample vendor */
+    usetup.id.product = 0x5678; /* sample product */
+    strcpy(usetup.name, "Example device");
+
+    ioctl(fd, UI_DEV_SETUP, &usetup);
+    ioctl(fd, UI_DEV_CREATE);
+
+    sleep(1);
+
+    // Print the parsed arguments
+    UT_LOG_DEBUG("Profile file: %s\n", profile_file);
+    UT_LOG_DEBUG("type: %s\n", type);
+    UT_LOG_DEBUG("code: %s\n", code);
+    UT_LOG_DEBUG("value: %d\n", value);
+
+    /* Key press, report the event, send key release, and report again */
+    emit(fd, get_type(pInstance, type), get_code(pInstance, type, code), value);
+
+    // /*
+    //  * Give userspace some time to read the events before we destroy the
+    //  * device with UI_DEV_DESTROY.
+    //  */
+    sleep(1);
```

Comment:
you can't have a sleep... You have to wait on an event or something else , need exact not..

I expect to be able to send keys back to back with ms appart,